From 250cda2614351358859a004824c8689e454a91a0 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Mon, 1 Dec 2014 08:06:49 -0800
Subject: [PATCH] Fix order of add_llvm_external_project calls.

---
 projects/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/projects/CMakeLists.txt b/projects/CMakeLists.txt
index 8d244fd..5aaf03f 100644
--- a/projects/CMakeLists.txt
+++ b/projects/CMakeLists.txt
@@ -19,8 +19,8 @@ if(${LLVM_BUILD_RUNTIME})
   # MSVC isn't quite working with libc++ yet, disable it until issues are
   # fixed.
   if(NOT MSVC)
-    add_llvm_external_project(libcxx)
     add_llvm_external_project(libcxxabi)
+    add_llvm_external_project(libcxx)
   endif()
   if(NOT LLVM_BUILD_EXTERNAL_COMPILER_RT)
     add_llvm_external_project(compiler-rt)
-- 
2.2.0

