From e24fe0eee01e9218bdb40e82c1238d99ac2b4bf5 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sun, 26 Oct 2014 12:18:44 -0700
Subject: [PATCH] Adding $prefix/include to list of standard include paths.

---
 lib/Driver/Tools.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index f1cbf50..ac345ab 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -449,6 +449,11 @@ void Clang::AddPreprocessingOptions(Compilation &C,
   // OBJCPLUS_INCLUDE_PATH - system includes enabled when compiling ObjC++.
   addDirectoryList(Args, CmdArgs, "-objcxx-isystem", "OBJCPLUS_INCLUDE_PATH");
 
+  if (!Args.hasArg(options::OPT_nostdlibinc)) {
+    CmdArgs.push_back("-internal-isystem");
+    CmdArgs.push_back(Args.MakeArgString(D.Dir + "/../include"));
+  }
+
   // Add C++ include arguments, if needed.
   if (types::isCXX(Inputs[0].getType()))
     getToolChain().AddClangCXXStdlibIncludeArgs(Args, CmdArgs);
-- 
1.9.3

