From bb59a887dd1c351bf79747f6023bb63161687607 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Mon, 20 Aug 2012 22:27:23 -0700
Subject: [PATCH] Adding a PREFIX option to change the dylib's install prefix.

---
 lib/buildit | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/lib/buildit b/lib/buildit
index 5938a92..9162952 100755
--- a/lib/buildit
+++ b/lib/buildit
@@ -22,6 +22,11 @@ then
     CC=clang
 fi
 
+if [ -z "$PREFIX" ]
+then
+	PREFIX=/usr
+fi
+
 if [ -z $MACOSX_DEPLOYMENT_TARGET ]
 then
 	if [ -z $IPHONEOS_DEPLOYMENT_TARGET ]
@@ -52,20 +57,20 @@ case $TRIPLE in
 		LDSHARED_FLAGS="-o libc++.1.dylib \
 			-dynamiclib -nodefaultlibs -current_version 1 \
 			-compatibility_version 1 \
-			-install_name /usr/lib/libc++.1.dylib \
-			-Wl,-reexport_library,/usr/lib/libc++abi.dylib \
+			-install_name $PREFIX/lib/libc++.1.dylib \
+			-Wl,-reexport_library,$PREFIX/lib/libc++abi.dylib \
 			-Wl,-unexported_symbols_list,libc++unexp.exp  \
 			/usr/lib/libSystem.B.dylib"
 	else
-		RE_EXPORT_LINE="/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++abi.exp"
+		RE_EXPORT_LINE="$PREFIX/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++abi.exp"
 		if [ -n "$SDKROOT" ]
 		then
 			EXTRA_FLAGS+="-isysroot ${SDKROOT}"
 			if echo "${RC_ARCHS}" | grep -q "armv7"  
 			then
-				RE_EXPORT_LINE="${SDKROOT}/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++sjlj-abi.exp"
+				RE_EXPORT_LINE="$PREFIX/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++sjlj-abi.exp"
 			else
-				RE_EXPORT_LINE="-Wl,-reexport_library,${SDKROOT}/usr/lib/libc++abi.dylib"
+				RE_EXPORT_LINE="-Wl,-reexport_library,$PREFIX/lib/libc++abi.dylib"
 			fi
 			CXX=`xcrun -sdk "${SDKROOT}"  -find clang++`
 			CC=`xcrun -sdk "${SDKROOT}"  -find clang`
@@ -74,7 +79,7 @@ case $TRIPLE in
 			-dynamiclib -nodefaultlibs  \
 			-current_version ${RC_ProjectSourceVersion} \
 			-compatibility_version 1 \
-			-install_name /usr/lib/libc++.1.dylib \
+			-install_name $PREFIX/lib/libc++.1.dylib \
 			-lSystem  \
 			-Wl,-unexported_symbols_list,libc++unexp.exp  \
 			${RE_EXPORT_LINE}  \
-- 
1.8.0.2

