From 43e8272c07c10f1df4bded1eb80e6f5826ce66b2 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Tue, 9 Oct 2012 07:51:57 -0700
Subject: [PATCH] Removing chown.

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 182dcc3..7856565 100644
--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,7 @@ installheaders::
 	mkdir -p $(HEADER_DIR)/c++/v1/ext
 	rsync -r --exclude=".*" --exclude="support" $(SRCDIRS)/include/* \
 	  $(HEADER_DIR)/c++/v1/
-	chown -R root:wheel $(HEADER_DIR)/c++
+	# chown -R root:wheel $(HEADER_DIR)/c++
 	chmod 755 $(HEADER_DIR)/c++/v1
 	chmod 644 $(HEADER_DIR)/c++/v1/*
 	chmod 755 $(HEADER_DIR)/c++/v1/ext
-- 
1.7.6

