From b7d2037c13fdb90364e71c95e229cf17543b7e66 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sat, 19 Jan 2013 11:02:57 -0800
Subject: [PATCH 2/2] Adding -L<prefix>/lib to default linker arguments.

---
 lib/Driver/Tools.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index e37959b..d68e746 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -187,6 +187,8 @@ static void AddLinkerInputs(const ToolChain &TC,
 
   // LIBRARY_PATH - included following the user specified library paths.
   addDirectoryList(Args, CmdArgs, "-L", "LIBRARY_PATH");
+  CmdArgs.push_back(Args.MakeArgString("-L" + D.InstalledDir + "/../lib"));
+  CmdArgs.push_back(Args.MakeArgString("-rpath=" + D.InstalledDir + "/../lib"));
 }
 
 /// \brief Determine whether Objective-C automated reference counting is
-- 
1.7.11.7

