From 8f3a646c971598b9f42dfc9a0de446887db7ed02 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sat, 27 Jul 2019 03:35:48 +0000
Subject: [PATCH 3/6] Adapting library's install_name on macOS to installation
 path.

---
 libcxxabi/src/CMakeLists.txt | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/libcxxabi/src/CMakeLists.txt b/libcxxabi/src/CMakeLists.txt
index 45d4d025361..f91f3f4af4d 100644
--- a/libcxxabi/src/CMakeLists.txt
+++ b/libcxxabi/src/CMakeLists.txt
@@ -102,13 +102,15 @@ if ( APPLE )
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
       "-current_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib"
+      "-nodefaultlibs")
     list(APPEND LIBCXXABI_LINK_FLAGS
         "/usr/lib/libSystem.B.dylib")
   else()
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib"
+      "-nodefaultlibs")
   endif()
 
   if (LLVM_USE_SANITIZER)
@@ -177,7 +179,9 @@ if (LIBCXXABI_ENABLE_SHARED)
                           SOVERSION
                             "1"
                           VERSION
-                            "1.0")
+                            "1.0"
+                          INSTALL_NAME_DIR
+                            "${CMAKE_INSTALL_PREFIX}/lib")
 
   if(LIBCXXABI_ENABLE_PIC)
     set_target_properties(cxxabi_shared PROPERTIES POSITION_INDEPENDENT_CODE ON)
-- 
2.22.0

