From eee3fa823a2649d7f07a2f6a938d1ecfa411d46b Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sun, 10 Mar 2019 11:01:27 -0700
Subject: [PATCH] Adapting library's install_name on macOS to installation
 path.

---
 lib/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 24489e8fb..7db3d4e64 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -160,7 +160,7 @@ if (LIBCXX_OSX_REEXPORT_LIBCXXABI_SYMBOLS)
     endif()
     add_link_flags(
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++.1.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++.1.dylib"
       "-Wl,-unexported_symbols_list,\"${CMAKE_CURRENT_SOURCE_DIR}/libc++unexp.exp\""
       "-Wl,-reexported_symbols_list,\"${RE_EXPORT_LIST}\""
       "-Wl,-force_symbols_not_weak_list,\"${CMAKE_CURRENT_SOURCE_DIR}/notweak.exp\""
@@ -248,6 +248,7 @@ if (LIBCXX_ENABLE_SHARED)
       OUTPUT_NAME   "c++"
       VERSION       "${LIBCXX_ABI_VERSION}.0"
       SOVERSION     "${LIBCXX_ABI_VERSION}"
+      INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
   )
   list(APPEND LIBCXX_BUILD_TARGETS "cxx_shared")
   if (LIBCXX_INSTALL_SHARED_LIBRARY)
-- 
2.21.0

