From b3ed22f8d142f9363036e5dd621ce9a5e5309e76 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sun, 10 Mar 2019 11:04:42 -0700
Subject: [PATCH] Adapting library's install_name on macOS to installation
 path.

---
 src/CMakeLists.txt | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7e21fb2..067d3e0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -94,13 +94,15 @@ if ( APPLE )
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
       "-current_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib"
+      "-nodefaultlibs")
     list(APPEND LIBCXXABI_LINK_FLAGS
         "/usr/lib/libSystem.B.dylib")
   else()
     list(APPEND LIBCXXABI_SHARED_LINK_FLAGS
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib"
+      "-nodefaultlibs")
   endif()
 
   if (LLVM_USE_SANITIZER)
@@ -178,7 +180,9 @@ if (LIBCXXABI_ENABLE_SHARED)
                           SOVERSION
                             "1"
                           VERSION
-                            "1.0")
+                            "1.0"
+                          INSTALL_NAME_DIR
+                            "${CMAKE_INSTALL_PREFIX}/lib")
   list(APPEND LIBCXXABI_BUILD_TARGETS "cxxabi_shared")
   if (LIBCXXABI_INSTALL_SHARED_LIBRARY)
     list(APPEND LIBCXXABI_INSTALL_TARGETS "cxxabi_shared")
-- 
2.21.0

