From dc13808356400492272a062fc8fa0e445d11d54a Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Tue, 29 Jul 2014 16:42:51 -0700
Subject: [PATCH] Adapting library's install_name on Darwin to installation
 prefix.

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a3fdf6c..1a4d024 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -67,12 +67,12 @@ if ( APPLE )
     list(APPEND link_flags
       "-compatibility_version 1"
       "-current_version ${LIBCXXABI_VERSION}"
-      "-install_name /usr/lib/libc++abi.1.dylib"
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib"
       "/usr/lib/libSystem.B.dylib")
   else()
     list(APPEND link_flags
       "-compatibility_version 1"
-      "-install_name /usr/lib/libc++abi.1.dylib")
+      "-install_name ${CMAKE_INSTALL_PREFIX}/lib/libc++abi.1.dylib")
   endif()
 endif()
 
-- 
1.9.3

