From 70f15c92e786e27e94b0db95a5a335513997163a Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Sat, 25 Oct 2014 15:07:27 -0700
Subject: [PATCH] Install unwind headers into $prefix/include.

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6815b0d..60bb67e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -255,3 +255,5 @@ if(NOT LIBCXXABI_ENABLE_SHARED)
 else()
   add_subdirectory(test)
 endif()
+
+install(FILES include/libunwind.h include/unwind.h DESTINATION include/)
-- 
1.9.3

