From 447bcc64597d67f40448886def55cb3a4cf5eef0 Mon Sep 17 00:00:00 2001
From: Matthias Vallentin <vallentin@icir.org>
Date: Wed, 18 Dec 2013 11:25:25 -0800
Subject: [PATCH] Create a static library on UNIX.

---
 lib/buildit |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/lib/buildit b/lib/buildit
index e4db411..3e802b9 100755
--- a/lib/buildit
+++ b/lib/buildit
@@ -89,12 +89,21 @@ set -x
 for FILE in ../src/*.cpp; do
 	$CXX -c -g -O3 $RC_CFLAGS $EXTRA_FLAGS -I../include $OPTIONS $FILE
 done
+
 case $TRIPLE in
   *-*-mingw*)
   for FILE in ../src/support/win32/*.cpp; do
     $CXX -c -g -Os $RC_CFLAGS $EXTRA_FLAGS -I../include $OPTIONS $FILE
   done
   ;;
+  *-apple-*)
+    # On Mac, we do not need to build a static library and can use
+    # -rrexport_library instead.
+  ;;
+  *)
+    ar cr libc++abi.a *.o
+    ranlib libc++abi.a
+  ;;
 esac
 $CC *.o $RC_CFLAGS $LDSHARED_FLAGS $EXTRA_FLAGS
 
-- 
1.7.6.4

