From 78a458155ab28fde90120ca35c8b7695d61a6ffe Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Mon, 12 Aug 2013 21:51:55 -0700
Subject: [PATCH] Removing cxx_demangle.h from expected libc++abi headers.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee5ef14..d78a972 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,7 @@ if ("${LIBCXX_CXX_ABI}" STREQUAL "libsupc++")
     )
 elseif ("${LIBCXX_CXX_ABI}" STREQUAL "libcxxabi")
   setup_abi_lib("LIBCXX_LIBCXXABI_INCLUDE_PATHS" ""
-    "c++abi" "cxxabi.h;cxa_demangle.h" ""
+    "c++abi" "cxxabi.h" ""
     )
 elseif ("${LIBCXX_CXX_ABI}" STREQUAL "libcxxrt")
   setup_abi_lib("LIBCXX_LIBCXXRT_INCLUDE_PATHS" "-DLIBCXXRT"
-- 
1.8.3.4

