From bcdbbe737fdba7fc08b71d53f3c8ba6860c217b7 Mon Sep 17 00:00:00 2001
From: Robin Sommer <robin@icir.org>
Date: Thu, 7 Jun 2018 10:04:27 -0700
Subject: [PATCH 2/2] Fix linker error on MacOS that was leading to undefined
 symbols.

---
 tools/debugserver/source/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/debugserver/source/CMakeLists.txt b/tools/debugserver/source/CMakeLists.txt
index e70e90923..31d8a86ce 100644
--- a/tools/debugserver/source/CMakeLists.txt
+++ b/tools/debugserver/source/CMakeLists.txt
@@ -8,6 +8,7 @@ include_directories(MacOSX)
 #include_directories(${CMAKE_CURRENT_BINARY_DIR}/MacOSX)
 
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -Wl,-sectcreate,__TEXT,__info_plist,${CMAKE_CURRENT_SOURCE_DIR}/../resources/lldb-debugserver-Info.plist")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-undefined,dynamic_lookup")
 
 check_cxx_compiler_flag("-Wno-gnu-zero-variadic-macro-arguments"
                         CXX_SUPPORTS_NO_GNU_ZERO_VARIADIC_MACRO_ARGUMENTS)
-- 
2.17.0

