From 436b8035fb861ebf93e3b556783f1956df8e4800 Mon Sep 17 00:00:00 2001
From: Matthias Vallentin <vallentin@icir.org>
Date: Wed, 5 Nov 2014 13:32:12 -0800
Subject: Remove hard-coded /usr/include/c++/v1 on FreeBSD.

---
 lib/Driver/ToolChains.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index b46f69d..6a6b750 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -2560,8 +2560,11 @@ void FreeBSD::AddClangCXXStdlibIncludeArgs(const ArgList &DriverArgs,
 
   switch (GetCXXStdlibType(DriverArgs)) {
   case ToolChain::CST_Libcxx:
-    addSystemInclude(DriverArgs, CC1Args,
-                     getDriver().SysRoot + "/usr/include/c++/v1");
+    // Unless we install into /usr/include, this entry will shadow
+    // PREFIX/include/c++/v1 which gets added somwhere down the line,
+    // so let's ignore the hard-coded prefix here. (Matthias Vallentin)
+    //addSystemInclude(DriverArgs, CC1Args,
+    //                 getDriver().SysRoot + "/usr/include/c++/v1");
     break;
   case ToolChain::CST_Libstdcxx:
     addSystemInclude(DriverArgs, CC1Args,
-- 
2.1.2

